### DISCUSSION TITLE
Does photoprism cleanup the cache if originals were moved ?

### DISCUSSION THREAD

**Discussion by nipunarora** [photoprism/photoprism#2402] [Q&A]
Does photoprism cleanup the cache if originals were moved? I need to restructure originals which were previously indexed. My workflow usually is to have a cron job do indexing, and I add photos to my originals folders directly, but sometimes I need to do restructuring, I'ld like to understand if I do reorganize my photos which are already indexed, would the previous thumbnails be cleaned up, and new ones be generated or something along those lines?

**Comment by graciousgrey** [comment]
If you index your originals folder and afterwards delete files from it or move files, PhotoPrism detects those changes during the next indexing. There is no need to regenerate thumbnails as those are stored in the cache within folders based on the filehash.

Avoid making those changes while the indexer is running!

Related: https://github.com/photoprism/photoprism/discussions/2319
